src/Proxy/ProxyGenerator.php

```diff
@@ -1124,6 +1128,17 @@ private function getParameterDefaultValue(ReflectionParameter $parameter)
 
         $value = rtrim(substr(explode('$' . $parameter->getName() . ' = ', (string) $parameter, 2)[1], 0, -2));
 
+        if (strpos($value, '\\') !== false || strpos($value, '::') !== false) {
+            $value = preg_split("/('(?:[^'\\\\]*+(?:\\\\.)*+)*+')/", $value, -1, PREG_SPLIT_DELIM_CAPTURE);
+            foreach ($value as $i => $part) {
+                if ($i % 2 === 0) {
+                    $value[$i] = preg_replace('/(?<![a-zA-Z0-9_\x7f-\xff\\\\])[a-zA-Z0-9_\x7f-\xff]++(?:\\\\[a-zA-Z0-9_\x7f-\xff]++|::)++/', '\\\\\0', $part);
```

Comment:
I think lines 1131, 1134 and 1135 warrant explanations/code comments, I can't say I understand what they do. I've tried dumping `$value` on line 1133 and it does not seem to do any splitting in the case of the constant. Also, removing the dot in the regex on line 1132 produces no effect.

Reply:
@nicolas-grekas maybe you can help? What should the `preg_split` part produce? Why are we acting only on `even` items of the array it produces?